### PR TITLE
fix: update mytarget

### PR DESCRIPTION
**Pull request by tom-grivaud**
### Description
- [x] Move the date check function to make it work
- [x] In rare cases the API indicates that it can't perform the request now, added a loop to retry in case the result is not fitting our expectations.

### Commits

- [x] Move the check date function to make it work.
- [x] Add a loop while the response doesn't contain the elements required.

### CODE REVIEW

File: nck/readers/mytarget_reader.py [hunk 1]
```diff
@@ -137,12 +136,21 @@ def __get_all_results(self, name_content: str, offset=0) -> List[Dict[str, str]]
...
+            for offset in range_offset:
+                resp = self.__get_response(name_content, offset=offset)
+                count = 0
+                while 'items' not in resp.keys() and count < 10:
+                    resp = self.__get_response(name_content, offset=offset)
+                    count += 1
+                if 'items' in resp.keys():
+                    elements.append(resp['items'])
+                else:
+                    logging.warning('Incorrect response from the API' + str(resp))
```
Comment (bibimorlet) [inline nck/readers/mytarget_reader.py:153]: Don't you want to use something like the **"retry"** decorator used in:
-dcm_client
-facebook_reader
-dbm_reader
...
?

Comment (bibimorlet) [reply nck/readers/mytarget_reader.py:153]: for example [here](https://github.com/artefactory/nautilus-connectors-kit/blob/630ba9ce882391c2d722e535e6b92c8fb4c2bfa7/nck/readers/dbm_reader.py#L153-L163) in dbm_reader

**Comment by tom-grivaud** [comment]
@bibimorlet Let me know if it's good to go for you, works perfectly on my side.

**Review by bibimorlet — APPROVED** [review]
👍

**Pull request merged**